Assets/Scripts/HPPickupInteract.cs

```diff
@@ -0,0 +1,20 @@
+using UnityEngine;
+
+public class HPPickupInteract : MonoBehaviour
+{
+    // Private Variables \\
+    [SerializeField]
+    private int hpPickupVal { get; set; } = 100;
+
+    // Methods \\
+    public void OnTriggerEnter2D(Collider2D coll2D)
```

Comment:
We might want to use `OnTriggerStay2D` ([documentation](https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnTriggerStay2D.html)) for this method. Currently, if the player loses health after being at max health while touching a health pickup, the health pickup is not applied until the player exits and enters the health pickup collider.

Reply:
Using `OnTriggerStay2D` does not have the behavior I thought it would. It seems like the player needs to move for that method to be called. I don't think it's important to change this right now. The health pickup still has its main functionality.